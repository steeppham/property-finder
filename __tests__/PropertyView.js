import 'react-native';
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import PropertyView from '../PropertyView.js';

describe('PropertyView', () => {
  it('should render PropertyView', () => {
    const property = {
      bedroom_number: '1',
      property_type: 'house',
      bathroom_number: '1',
      img_url: 'http://someimage.com/image.jpg',
      title: 'title',
      summary: 'summary',
      price_formatted: '$123 GP'
    }
    const component = shallow(<PropertyView property={property}/>);
    expect(component).toMatchSnapshot();
  });
});