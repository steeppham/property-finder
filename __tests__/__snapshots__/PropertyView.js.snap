// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyView should render PropertyView 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "marginTop": 65,
          }
    }
>
    <Image
        source={
            Object {
                "uri": "http://someimage.com/image.jpg",
              }
        }
        style={
            Object {
                "height": 300,
                "width": 400,
              }
        }
    />
    <View
        style={
            Object {
                "backgroundColor": "#F8F8F8",
              }
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#48BBEC",
                    "fontSize": 25,
                    "fontWeight": "bold",
                    "margin": 5,
                  }
            }
        >
            $123
        </Text>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#656565",
                    "fontSize": 20,
                    "margin": 5,
                  }
            }
        >
            title
        </Text>
        <View
            style={
                Object {
                    "backgroundColor": "#DDDDDD",
                    "height": 1,
                  }
            }
        />
    </View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#656565",
                "fontSize": 18,
                "margin": 5,
              }
        }
    >
        1 bed house, 1 bathroom
    </Text>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "#656565",
                "fontSize": 18,
                "margin": 5,
              }
        }
    >
        summary
    </Text>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "marginTop": 65,
                }
      }
>
      <Image
            source={
                  Object {
                        "uri": "http://someimage.com/image.jpg",
                      }
            }
            style={
                  Object {
                        "height": 300,
                        "width": 400,
                      }
            }
      />
      <View
            style={
                  Object {
                        "backgroundColor": "#F8F8F8",
                      }
            }
      >
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#48BBEC",
                              "fontSize": 25,
                              "fontWeight": "bold",
                              "margin": 5,
                            }
                  }
            >
                  $123
            </Text>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#656565",
                              "fontSize": 20,
                              "margin": 5,
                            }
                  }
            >
                  title
            </Text>
            <View
                  style={
                        Object {
                              "backgroundColor": "#DDDDDD",
                              "height": 1,
                            }
                  }
            />
      </View>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "#656565",
                        "fontSize": 18,
                        "margin": 5,
                      }
            }
      >
            1 bed house, 1 bathroom
      </Text>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "#656565",
                        "fontSize": 18,
                        "margin": 5,
                      }
            }
      >
            summary
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PropertyView
        property={
                Object {
                        "bathroom_number": "1",
                        "bedroom_number": "1",
                        "img_url": "http://someimage.com/image.jpg",
                        "price_formatted": "$123 GP",
                        "property_type": "house",
                        "summary": "summary",
                        "title": "title",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PropertyView {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "property": Object {
            "bathroom_number": "1",
            "bedroom_number": "1",
            "img_url": "http://someimage.com/image.jpg",
            "price_formatted": "$123 GP",
            "property_type": "house",
            "summary": "summary",
            "title": "title",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "marginTop": 65,
                            }
          }
>
          <Image
                    source={
                              Object {
                                        "uri": "http://someimage.com/image.jpg",
                                      }
                    }
                    style={
                              Object {
                                        "height": 300,
                                        "width": 400,
                                      }
                    }
          />
          <View
                    style={
                              Object {
                                        "backgroundColor": "#F8F8F8",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#48BBEC",
                                                  "fontSize": 25,
                                                  "fontWeight": "bold",
                                                  "margin": 5,
                                                }
                              }
                    >
                              $123
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#656565",
                                                  "fontSize": 20,
                                                  "margin": 5,
                                                }
                              }
                    >
                              title
                    </Text>
                    <View
                              style={
                                        Object {
                                                  "backgroundColor": "#DDDDDD",
                                                  "height": 1,
                                                }
                              }
                    />
          </View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#656565",
                                        "fontSize": 18,
                                        "margin": 5,
                                      }
                    }
          >
                    1 bed house, 1 bathroom
          </Text>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#656565",
                                        "fontSize": 18,
                                        "margin": 5,
                                      }
                    }
          >
                    summary
          </Text>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "marginTop": 65,
                            }
          }
>
          <Image
                    source={
                              Object {
                                        "uri": "http://someimage.com/image.jpg",
                                      }
                    }
                    style={
                              Object {
                                        "height": 300,
                                        "width": 400,
                                      }
                    }
          />
          <View
                    style={
                              Object {
                                        "backgroundColor": "#F8F8F8",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#48BBEC",
                                                  "fontSize": 25,
                                                  "fontWeight": "bold",
                                                  "margin": 5,
                                                }
                              }
                    >
                              $123
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#656565",
                                                  "fontSize": 20,
                                                  "margin": 5,
                                                }
                              }
                    >
                              title
                    </Text>
                    <View
                              style={
                                        Object {
                                                  "backgroundColor": "#DDDDDD",
                                                  "height": 1,
                                                }
                              }
                    />
          </View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#656565",
                                        "fontSize": 18,
                                        "margin": 5,
                                      }
                    }
          >
                    1 bed house, 1 bathroom
          </Text>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "#656565",
                                        "fontSize": 18,
                                        "margin": 5,
                                      }
                    }
          >
                    summary
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PropertyView
    property={
        Object {
            "bathroom_number": "1",
            "bedroom_number": "1",
            "img_url": "http://someimage.com/image.jpg",
            "price_formatted": "$123 GP",
            "property_type": "house",
            "summary": "summary",
            "title": "title",
          }
    }
/>,
}
`;
